### PR TITLE
Pycharm Support (including debugging in Docker)

### PR DESCRIPTION
**Pull request by noisy**
New implementation of https://github.com/pydanny/cookiecutter-django/pull/306

* [x] proper settings in _Language & Frameworks > Django_
* [x] Default Python Template Language set to Django instead of Jinja2 in _Language & Frameworks > Python Template Language_
* [x] `node_modules` directory added to excluded (will not be indexed)
* [x] Example Run/Debug Configurations for runserver, runserver_pluse, migrate
* [x] Example Run/Debug Configurations for tests (all, from module, from file, from class, specific)
* [x] `requirements/local.txt` marked as default for development in _Tools > Python Integreated Tools_
* [x] documentation - [pycharm/configuration.rst](https://github.com/SpisTresci/cookiecutter-django/blob/pycharm_support_2/%7B%7Bcookiecutter.repo_name%7D%7D/docs/pycharm/configuration.rst)

### CODE REVIEW

File: {{cookiecutter.repo_name}}/.gitignore [hunk 1]
```diff
@@ -37,8 +37,10 @@ htmlcov
 *.mo
 *.pot
 
+{% if cookiecutter.use_pycharm == 'y' %}
```
Comment (luzfcb) [inline {{cookiecutter.repo_name}}/.gitignore]: I ignore the .idea directory even if not use PyCharm.

**Comment by jayfk** [comment]
screen shot 2016-03-29 at 17 07 34

Any idea why they are marked red?

**Comment by noisy** [comment]
@jayfk probably you didn't read: [pycharm/configuration.rst](https://github.com/SpisTresci/cookiecutter-django/blob/pycharm_support_2/%7B%7Bcookiecutter.repo_name%7D%7D/docs/pycharm/configuration.rst). You have to manually setup docker, like it is described in provided documentation.

**Comment by jayfk** [comment]
Yeah, I have read that:

screen shot 2016-03-29 at 17 14 00

The commands are running tough.

**Comment by jayfk** [comment]
This LGTM. :+1:

**Comment by pydanny** [comment]
Not being the Docker or PyCharm expert, I nevertheless think this looks good. Anyone else want to pitch in? :bell: @luzfcb @theskumar @burhan @pauleveritt

**Comment by burhan** [comment]
I'm here at DjangoConEU and Jetbrains guys responsible for PyCharm are here as sponsors - let me run this by them and see if they can see any issues with it.

**Comment by pauleveritt** [comment]
I’m behind on paying attention to this ticket, but it looks like things have happened a lot on both sides in the last 3 months. FWIW, with the new Docker stuff that is in beta, it will be a lot easier for JetBrains/PyCharm to work on this stuff. It’s been hard on our team…lots of VirtualBox and storage-related heisenbugs.

—Paul

**Comment by burhan** [comment]
@Paul - with the native integration of Docker into Windows and OSX, you'll probably need to revisit the integration again soon and hopefully that should be death of docker-machine and virtualbox headaches.

**Comment by luzfcb** [comment]
I probably have a little free time to test it on next Sunday.

**Comment by jayfk** [comment]
I think we can merge this now. If there are errors on windows, we'll get them sorted out. Any objections?

**Comment by Braintelligence** [comment]
I'm sorry for derailing this, but I hope someone who's participating in this pull can give me a hint on this:

When I change the base.txt of the requirements.txt files, so as to add a package to my remote docker interpreter, it doesn't get added the next time I run a docker configuration in PyCharm.

I guess I have to rebuild the container for this, but I don't know the most comfortable way to do this. Is this somehow baked into PyCharm?

**Comment by pydanny** [comment]
@braintelligence, please don't hijack (derail) this ticket. 

1. If you have identified a bug, open a separate issue. 
2. If you have a question, ask in gitter or on stack overflow.

**Pull request merged**